mesonbuild/modules/gnome.py

```diff
@@ -1016,55 +1017,73 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs) ->
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = mesonlib.stringlistify(kwargs.pop('languages', []))
+        if not langs:
+            langs = read_linguas(state.subdir)
+        # I am currently ignoring media and symlinks
+        # How could they be added?
         media = mesonlib.stringlistify(kwargs.pop('media', []))
         symlinks = kwargs.pop('symlink_media', True)
+        targets = []
 
         if not isinstance(symlinks, bool):
             raise MesonException('symlink_media must be a boolean')
 
         if kwargs:
             raise MesonException('Unknown arguments passed: {}'.format(', '.join(kwargs.keys())))
 
-        script = state.environment.get_build_command()
-        args = ['--internal',
-                'yelphelper',
-                'install',
-                '--subdir=' + state.subdir,
-                '--id=' + project_id,
-                '--installdir=' + os.path.join(state.environment.get_datadir(), 'help'),
-                '--sources=' + source_str]
-        if symlinks:
-            args.append('--symlinks=true')
-        if media:
-            args.append('--media=' + '@@'.join(media))
-        if langs:
-            args.append('--langs=' + '@@'.join(langs))
-        inscript = state.backend.get_executable_serialisation(script + args)
-
-        potargs = state.environment.get_build_command() + [
-            '--internal', 'yelphelper', 'pot',
-            '--subdir=' + state.subdir,
-            '--id=' + project_id,
-            '--sources=' + source_str,
-        ]
-        pottarget = build.RunTarget('help-' + project_id + '-pot', potargs,
-                                    [], state.subdir, state.subproject)
-
-        poargs = state.environment.get_build_command() + [
-            '--internal', 'yelphelper', 'update-po',
-            '--subdir=' + state.subdir,
-            '--id=' + project_id,
-            '--sources=' + source_str,
-            '--langs=' + '@@'.join(langs),
-        ]
-        potarget = build.RunTarget('help-' + project_id + '-update-po', poargs,
-                                   [], state.subdir, state.subproject)
-
-        rv = [inscript, pottarget, potarget]
-        return ModuleReturnValue(None, rv)
+        pot_name = project_id + '.pot'
+        pot_kwargs = {'command': ['itstool', '-o', '@OUTPUT@', '@INPUT@'],
+                      'input': sources,
+                      'output': pot_name,
+        }
+        pottarget = build.CustomTarget('help-' + project_id + '-pot',
+                                       os.path.join(state.subdir, 'C'),
+                                       state.subproject, pot_kwargs)
+        targets.append(pottarget)
+        pot_file = mesonlib.File.from_built_file(os.path.join(state.subdir, 'C'),
+                                                 pot_name)
```

Comment:
Mmm, yes, I saw that in your changes and also something similar in `gtkdochelper.py`, but I couldn't really understand why. Now you mentioning a ".PHONY" target, I understand that, as there exists no `.pot` file to begin with, this is not a rule associated with a file and should be ".PHONY". However, some questions:
- Is there some way to limit the execution of this target to only when the `input` files change? It is true the `.pot` file is not a source file, but what's the point of rebuilding documentation every time if the documentation files did not change? I think this might be a clear difference between the `i18n.gettext` and `gnome.yelp`, as the former defines the source files, whereas the first one basically looks at the whole tree (unless I'm getting something wrong...)
- Which is the recommended way to handle temporary non-source files like the `.pot` and `.po`? I honestly very much liked that the targets are more clear, short and readable than the helper script :)

Reply:
They aren't temporary :D they traditionally get checked into git.

The primary concern here about having them get built by input/output rules is that source files are used to build pot files, and pot files are used to build po files, and po files are used to build mo files.

But, you only want pot/po to be triggered by maintainers, as po files have *two* inputs. The pot file, and the contents pulled from e.g. transifex. And the mo files only use the transifex updates, not the ones merged from the pot file...

It is probably ok to have the pot file generated via a non-phony target, and just not built by default. However, po files should still be recreated every time the maintainer target is run... because we can't know whether the po files were updated from transifex and are more recent than the pot file, yet still need to merge pot updates too.

The pot -> po rules can still use direct gettext utility commands rather than going through a helper script, but they should still be .PHONY rules (RunTarget) I think.

Reply:
First of all, thank you very much for your time and patience, I really appreciate it!

I think I have gotten some concepts totally wrong here, based on my little and biased experience with translatable programs and lack of understanding of previous code. My understanding was that during build we would have:

  `source` + `.po` files in VCS -> generate `.pot` from `source` -> new `.po` files with `.pot` and VCS'ed `.po` -> generate `.mo` (or here `.gmo` + merge) -> install `.mo` (or merged documents here in yelp).
 
It seems I was only considering the packaging process and totally dismissing the maintainer process. Then after your comments, my understanding is:

- **Maintainer** process: (updated) `source` + `.po` files in VCS -> generate `.pot` from `source` -> new `.po` files with `.pot` and (updated) VCS'ed `.po` -> maintainer commits updates.
- **Packaging** process: `source` + `.po` files in VCS -> `.mo` (or here `.gmo` + merge) -> install `.mo` (or merged documents here in yelp)

If my understanding now is correct, it makes total sense to have both the `.pot` and the `.po` targets as RunTargets, as they are not needed for packaging, but might be used by developers when updating translations.

Reply:
Hopefully last push has fixed this. At least I have verified that running `meson compile help-{package_id}-pot` and `meson compile help-{package_id}-update-po` seem to work as expected.